From a55b6e8fec12672efcd02a048e98e66cd16a7603 Mon Sep 17 00:00:00 2001
From: Lionel Debieve <lionel.debieve@foss.st.com>
Date: Fri, 3 Feb 2023 08:49:38 +0100
Subject: [PATCH 0068/1141] arm64: dts: st: add arm-wdt node for watchdog
 support on stm32mp251

Add the node to use the ARM SMC watchdog support. It will
use a dedicated smc-id to configure and kick the watchdog.

Signed-off-by: Lionel Debieve <lionel.debieve@foss.st.com>
Change-Id: I08b49d5e2f9d666ceffa1f4d21b9eb36ac1028fd
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/288994
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -28,6 +28,12 @@
 		interrupt-parent = <&intc>;
 	};
 
+	arm_wdt: watchdog {
+		compatible = "arm,smc-wdt";
+		arm,smc-id = <0xb200005a>;
+		status = "disabled";
+	};
+
 	clocks {
 		ck_flexgen_08: ck-flexgen-08 {
 			#clock-cells = <0>;
